From 7c5af50b02b65bc2074116a54b12ba7964f9c3f1 Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Date: Wed, 4 Jan 2012 16:25:10 +0200
Subject: [PATCH 2/4] egl: Add NULL check for eglGetconfigs num_config
 parameter

EGL 1.4 specification says in 3.4.1:
"An EGL_BAD_PARAMETER error is generated if num config is NULL."

We have to check for the condition and return the required error to
application.

Change-Id: Ib42709f0420161ce661536394d4c0779bc62be57
Original-Change-Id: I4fe670c4362f4ae494f91d69ae105e29ccf3c769
Signed-off-by: Pauli Nieminen <pauli.nieminen@linux.intel.com>
---
 opengl/libagl/egl.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/opengl/libagl/egl.cpp b/opengl/libagl/egl.cpp
index c1efd1c..2d03454 100644
--- a/opengl/libagl/egl.cpp
+++ b/opengl/libagl/egl.cpp
@@ -1459,6 +1459,9 @@ EGLBoolean eglGetConfigs(   EGLDisplay dpy,
     if (egl_display_t::is_valid(dpy) == EGL_FALSE)
         return setError(EGL_BAD_DISPLAY, EGL_FALSE);

+    if (ggl_unlikely(num_config==0))
+        return setError(EGL_BAD_PARAMETER, EGL_FALSE);
+
     GLint numConfigs = NELEM(gConfigs);
     if (!configs) {
         *num_config = numConfigs;
--
2.7.4
